RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -48,61 +48,161 @@ namespace {
+    for (int i=2; i>=0; --i) {
+      drCut[i] = powN(par[i]*nLayers,4)*drE;
+    }
+  }
+  
+  inline void dzCut_par2(reco::Track const & trk, int & nLayers, const float * par, const float * d0err, const float * d0err_par, float dzCut[]) {
+    float pt = float(trk.pt());
+    float p  = float(trk.p());
+
+    for (int i=2; i>=0; --i) {
+      // parametrized d0 resolution for the track pt
+      float nomd0E = sqrt(d0err[i]*d0err[i]+(d0err_par[i]/pt)*(d0err_par[i]/pt));
+      // parametrized z0 resolution for the track pt and eta
+      //    float nomdzE = nomd0E*(std::cosh(eta));
+      float nomdzE = nomd0E*(p/pt); // cosh(eta):=abs(p)/pt
+      
+      dzCut[i] = powN(par[i]*nLayers,4)*nomdzE;
+    }
   }
+  inline void drCut_par2(reco::Track const & trk, int & nLayers, const float* par, const float * d0err, const float * d0err_par, float drCut[]) {
+    float pt = trk.pt();
+
+    for (int i=2; i>=0; --i) {
+      // parametrized d0 resolution for the track pt
+      float nomd0E = sqrt(d0err[i]*d0err[i]+(d0err_par[i]/pt)*(d0err_par[i]/pt));
 
+      drCut[i] = powN(par[i]*nLayers,4)*nomd0E;
+    }
+  }
+  
   struct Cuts {
     
     Cuts(const edm::ParameterSet & cfg) {
-      fillArrayF(maxChi2,cfg,"maxChi2");
-      fillArrayF(maxChi2n,cfg,"maxChi2n");
-      fillArrayI(minPixelHits,cfg,"minPixelHits");
-      fillArrayI(min3DLayers,cfg,"min3DLayers");
-      fillArrayI(minLayers,cfg,"minLayers");
+      fillArrayF(maxChi2,      cfg,"maxChi2");
+      fillArrayF(maxChi2n,     cfg,"maxChi2n");
+      fillArrayI(minPixelHits, cfg,"minPixelHits");
+      fillArrayI(min3DLayers,  cfg,"min3DLayers");
+      fillArrayI(minLayers,    cfg,"minLayers");
       fillArrayI(maxLostLayers,cfg,"maxLostLayers");
-      fillArrayF(maxDz,cfg,"maxDz");
-      fillArrayF(maxDr,cfg,"maxDr");
-
+      minNVtxTrk = cfg.getParameter<int>("minNVtxTrk");
+      fillArrayF(maxDz,        cfg,"maxDz");
+      fillArrayF(maxDzWrtBS,   cfg,"maxDzWrtBS");
+      fillArrayF(maxDr,        cfg,"maxDr");
+      edm::ParameterSet dz_par = cfg.getParameter<edm::ParameterSet>("dz_par");
+      fillArrayF(dz_par1,      dz_par,"dz_par1");
+      fillArrayF(dz_par2,      dz_par,"dz_par2");
+      edm::ParameterSet dr_par = cfg.getParameter<edm::ParameterSet>("dr_par");
+      fillArrayF(dr_par1,      dr_par,"dr_par1");
+      fillArrayF(dr_par2,      dr_par,"dr_par2");
+      fillArrayF(d0err,        dr_par,"d0err");
+      fillArrayF(d0err_par,    dr_par,"d0err_par");
     }
     
     
     
     float operator()(reco::Track const & trk,
-		   reco::BeamSpot const & beamSpot,
-		   reco::VertexCollection const & vertices,
-		   GBRForest const *) const {
-
-     
+		     reco::BeamSpot const & beamSpot,
+		     reco::VertexCollection const & vertices,
+		     GBRForest const *) const {
       
       float ret = 1.f;
+      float dummy[3] = {1E-5, 1E-5, 1E-5};
+      ret = std::min(ret,cut(float(trk.ndof()),dummy,std::greater_equal<float>()) );
```

Comment:
in the end, most of the differences were coming from here, from tracks with zero d.o.f.

I checked a couple and they had 2 pixel hits and 1 strip hit.
Is it fair to hardcode this cut? Shouldn't it at least be made configurable?